src/service/okapi.service.ts

```diff
@@ -32,36 +32,18 @@ class OkapiService extends RestService {
           let refreshToken;
           const headers = !!response?.headers ? response?.headers : undefined;
 
-          if (!!headers['set-cookie']) {
-            const matchAccess = /folioAccessToken=([^;\s]*)/gi;
-            const matchRefresh = /folioRefreshToken=([^;\s]*)/gi;
-            const cookies = !!headers['set-cookie'] ? headers['set-cookie'] : undefined;
-            let foundAccess;
-            let foundRefresh;
+          if (!!response.headers['set-cookie'] && Array.isArray(response.headers['set-cookie'])) {
+            const get = (token: string) => {
+              const maybe = response.headers['set-cookie'].filter((cookie: string) => cookie.startsWith(token));
 
-            if (!!cookies && Array.isArray(cookies)) {
-              let matched;
-
-              for (let i = 0; i < cookies.length; i++) {
-                matched = cookies[i].match(matchAccess);
-
-                if (!!matched) {
-                  foundAccess = matched;
-                } else {
-                  matched = cookies[i].match(matchRefresh);
-
-                  if (!!matched) {
-                    foundRefresh = matched;
-                  }
-                }
+              if (maybe.length > 0) {
+                return maybe[0].substring(token.length + 1, maybe[0].indexOf(';'));
```

Comment:
This string `folioAccessToken;` should not match but might (And also `folioAccessTokenBlahBlahBlah=...;`).

Please use more precise matching to at least make sure the format is like that described in the regex: `/folioAccessToken=([^;\s]*)/gi;`, such as `/$token=([^;\s]*)/gi;` (assuming the `$token` variable gets expanded appropriately in a regex statement).